src/components/modebar/manage.js

```diff
@@ -170,19 +170,42 @@ function getButtonGroups(gd) {
         dragModeGroup.push('select2d', 'lasso2d');
     }
 
-    // accept pre-defined buttons as string
+    var enabledHoverGroup = [];
+    var enableHover = function(a) {
+        // return if already added
+        if(enabledHoverGroup.indexOf(a) !== -1) return;
+        // should be in hoverGroup
+        if(hoverGroup.indexOf(a) !== -1) {
+            enabledHoverGroup.push(a);
+        }
+    };
```

Comment:
So what you're saying is, we still calculate the hover entries you would have gotten, back in 1.x when we included them by default, and in 2.x you won't get them by default but you'll be allowed to re-include any of those exact buttons but no others.

I guess it makes sense to restrict to only the buttons we know will work, but as a user it may be hard to figure out what those buttons are in any given case. For example Sankey gets `['hoverClosestCartesian', 'hoverCompareCartesian']`? And some multi-type plots just get `['toggleHover']` but others get something more specific?

What if we accept a special `buttonsToAdd` value `'hovermode'`, that copies all of `hoverGroup` into `enabledHoverGroup` - ie to get back 1.x behavior no matter what traces you have you can just use that one special value?

Reply:
> What if we accept a special buttonsToAdd value 'hovermode'

this sounds like a good solution to me.

Reply:
maybe `v1hovermode` or something to clearly identify this as "the v1 behaviour" ?

Reply:
> So what you're saying is, we still calculate the hover entries you would have gotten, back in 1.x when we included them by default, and in 2.x you won't get them by default but you'll be allowed to re-include any of those exact buttons but no others.
> 
Exactly.

Reply:
> What if we accept a special `buttonsToAdd` value `'hovermode'`, that copies all of `hoverGroup` into `enabledHoverGroup` - ie to get back 1.x behavior no matter what traces you have you can just use that one special value?
It is doable but at the same time rather confusing (and perhaps hard to maintain) as the v1 `hoverGroup` does include `spike` options too. Plus in that case we may restrict users to only include all of those buttons but not some?

Reply:
Right, if you want some subset of the v1 behavior you have to figure out the specific options you need and specify them explicitly.

Good point re: spikes. `'togglespikelines'` is pretty simple and easy to document; so let's have `'v1hovermode'` give you just the hovermode buttons - ie everything in `hoverGroup` except `toggleSpikelines` if it's there.

Reply:
OK.
How about `allhovermodes` instead of `v1hovermode`?
Later on the road we may want to add extra buttons e.g. for `unified hover` options and having `v1` would make it difficult to add those.

Reply:
If we call it `allhovermodes` it's wrong until we add unified. And unified is more a decision chart developers will make, whereas compare vs closest has times it's useful to viewers (mainly for charts that the developer didn't explicitly specify the best hovermode) so I don't see much need to add a button for unified.

Reply:
> If we call it allhovermodes it's wrong until we add unified

Yes. I would favour `v1hovermodes` as it's clear that this is simply brining back the old mode without making any additional promises/implying anything else about "all" anything :)

Reply:
Added in https://github.com/plotly/plotly.js/pull/5654/commits/f4e6120f8b4595ee44db9ce14971a1ad24a803c1.